src/components/ChannelList/hooks/listeners/useNewMessage.js

```diff
@@ -0,0 +1,20 @@
+import { useContext, useEffect } from 'react';
+
+import { ChatContext } from '../../../../context';
+import { moveChannelUp } from '../../utils';
+
+export const useNewMessage = ({ lockChannelOrder, setChannels }) => {
+  const { client } = useContext(ChatContext);
+
+  useEffect(() => {
+    const handleEvent = (e) => {
+      setChannels((channels) => {
+        if (!lockChannelOrder) return moveChannelUp(e.cid, channels);
```

Comment:
personally still don't love this single line no brackets if statement style. shortens the code but also I think it negatively effects readability a super small bit